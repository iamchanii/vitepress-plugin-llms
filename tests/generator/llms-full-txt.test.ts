import { describe, expect, it } from 'bun:test'

import {
	generateLLMsFullTxt,
	// @ts-ignore
} from '@/generator/llms-full-txt'
import { preparedFilesSample, sampleDomain } from '../resources'

describe('generateLLMsFullTxt', () => {
	it('generates a `llms-full.txt` file', async () => {
		expect(await generateLLMsFullTxt(preparedFilesSample.slice(1), {})).toMatchSnapshot()
	})

	it('correctly attaches the domain to URLs in context', async () => {
		expect(
			await generateLLMsFullTxt(preparedFilesSample.slice(1), {
				domain: sampleDomain,
			}),
		).toMatchSnapshot()
	})
})
