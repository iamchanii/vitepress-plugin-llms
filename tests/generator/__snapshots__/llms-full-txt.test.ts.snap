// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`generateLLMsFullTxt generates a \`llms-full.txt\` file 1`] = `
"---
url: /test/getting-started.md
description: Instructions on how to get started with the tool
---

# Installation

blah blah blah...

---

---
url: /test/quickstart.md
description: Instructions for quick project initialization
---

# Project initialization

blah blah blah...

---

---
url: /test/other.md
---
# Some cool stuff
"
`;

exports[`generateLLMsFullTxt correctly attaches the domain to URLs in context 1`] = `
"---
url: 'https://example.com/test/getting-started.md'
description: Instructions on how to get started with the tool
---

# Installation

blah blah blah...

---

---
url: 'https://example.com/test/quickstart.md'
description: Instructions for quick project initialization
---

# Project initialization

blah blah blah...

---

---
url: 'https://example.com/test/other.md'
---
# Some cool stuff
"
`;
